src/main/java/homework_6/map_problems_generator/MapProblemsMutableGenerator.java

```diff
@@ -0,0 +1,40 @@
+package homework_6.map_problems_generator;
+
+import java.util.Random;
+
+public class MapProblemsMutableGenerator {
+
+    private String value;
+
+    public MapProblemsMutableGenerator(String value) {
+        this.value = value;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj == null || obj.getClass() != this.getClass()) {
+            return false;
+        }
+
+        MapProblemsCollisionGenerator example = (MapProblemsCollisionGenerator) obj;
+
+        return (value != null && value.equals(example.getValue()));
+    }
+
+    @Override
+    public int hashCode() {
+        return new Random().nextInt(121);
+    }
```

Comment:
I know it's unclear from the homework description, but here is not a mutable key problem, it's just invalid hashcode impl. Mutable key problem appears when hashcode generation is valid